internal/services/network/virtual_network_peering_resource.go

```diff
@@ -0,0 +1,234 @@
+		}
+	}
+
+	peer := network.VirtualNetworkPeering{
+		Name:                                  &id.Name,
+		VirtualNetworkPeeringPropertiesFormat: getVirtualNetworkPeeringProperties(d),
+	}
+
+	peerMutex.Lock()
+	defer peerMutex.Unlock()
+
+	if err := pluginsdk.Retry(300*time.Second, retryVnetPeeringsClientCreateUpdate(d, id.ResourceGroup, id.VirtualNetworkName, id.Name, peer, meta)); err != nil {
+		return err
+	}
+
+	d.SetId(id.ID())
+
+	return virtualNetworkPeeringRead(d, meta)
+}
+
+func virtualNetworkPeeringRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.VnetPeeringsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VirtualNetworkPeeringID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.VirtualNetworkName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("making Read request on %s: %+v", *id, err)
+	}
+
+	// update appropriate values
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("name", id.Name)
+	d.Set("virtual_network_name", id.VirtualNetworkName)
+
+	if peer := resp.VirtualNetworkPeeringPropertiesFormat; peer != nil {
+		d.Set("allow_virtual_network_access", peer.AllowVirtualNetworkAccess)
+		d.Set("allow_forwarded_traffic", peer.AllowForwardedTraffic)
+		d.Set("allow_gateway_transit", peer.AllowGatewayTransit)
+		d.Set("use_remote_gateways", peer.UseRemoteGateways)
+		if network := peer.RemoteVirtualNetwork; network != nil {
+			d.Set("remote_virtual_network_id", network.ID)
+		}
+	}
+
+	return nil
+}
+
+func virtualNetworkPeeringDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.VnetPeeringsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VirtualNetworkPeeringID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	peerMutex.Lock()
+	defer peerMutex.Unlock()
+
+	future, err := client.Delete(ctx, id.ResourceGroup, id.VirtualNetworkName, id.Name)
+	if err != nil {
+		return fmt.Errorf("deleting %s: %+v", *id, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for deletion of %s: %+v", *id, err)
+	}
+
+	return err
+}
+
+func getVirtualNetworkPeeringProperties(d *pluginsdk.ResourceData) *network.VirtualNetworkPeeringPropertiesFormat {
+	allowVirtualNetworkAccess := d.Get("allow_virtual_network_access").(bool)
+	allowForwardedTraffic := d.Get("allow_forwarded_traffic").(bool)
+	allowGatewayTransit := d.Get("allow_gateway_transit").(bool)
+	useRemoteGateways := d.Get("use_remote_gateways").(bool)
+	remoteVirtualNetworkID := d.Get("remote_virtual_network_id").(string)
+
+	return &network.VirtualNetworkPeeringPropertiesFormat{
+		AllowVirtualNetworkAccess: &allowVirtualNetworkAccess,
```

Comment:
minor optional nit, these should all be
```suggestion
		AllowVirtualNetworkAccess: pointer.FromBool(d.Get("allow_virtual_network_access").(bool))
```
```suggestion
		AllowVirtualNetworkAccess: &allowVirtualNetworkAccess,
```